### PR TITLE
React & Redux group project - Space Travelers' Hub

### PR DESCRIPTION
**Pull request by seyioyemade**
## React & Redux group project - Space Travelers' Hub

In this milestone we:

- Configured & setup react and redux.
- Fetched data and updated Redux store.
- Rendered the data in the UI.
- Wrote actions and reducers for booking rockets and joining missions.
- Wrote actions and reducers for canceling rockets and leaving missions.
- Used conditional components rendering.
- Created My Profile section.
- Tested all components.

### CODE REVIEW

File: src/components/missions/MissionsLogic.jsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import MissionsTable from './MissionsTable';
+import { missionsData } from '../../redux/missions/MissionsAPI';
+
+const MissionsLogic = () => {
+  const { loading, missions, error } = useSelector((state) => state.missions);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(missionsData());
+  }, [dispatch]);
```
Comment (levy002) [inline src/components/missions/MissionsLogic.jsx:13]: - [ ] Kindly make sure the fetched data are not being added to the store whenever a user changes view / uses navigations.

    - ### APP BEHAVIOUR
    
    chrome-capture-2023-4-18 (3)

    _**Hint: Make sure to dispatch the `missionsData()` action only when the `missions`  array in the redux store is empty**_

File: src/components/rockets/Lists.jsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import { FetchData } from "../../redux/rockets/API"
+import {useDispatch, useSelector} from 'react-redux'
+import styles from "../../styles/rockets/rocket.module.css";
+
+export const Lists = ()=>{
+    const {rockets, status, error} = useSelector((state)=> state.rockets)
+    const Dispatch = useDispatch()
+    useEffect(()=>{
+        Dispatch(FetchData())
+    }, [Dispatch])
```
Comment (levy002) [inline src/components/rockets/Lists.jsx:12]: - [ ] Kindly make sure the fetched data are not being added to the store whenever a user changes view / uses navigations.

    - ### APP BEHAVIOUR
    
   chrome-capture-2023-4-18 (2)

  

    _**Hint: Make sure to dispatch the `FetchedData()` action only when the `rockets`  array in the redux store is empty**_

**Review by levy002 — CHANGES_REQUESTED** [review]
## Hello @seyioyemade & @AbbasSarwar   , :wave:
### I'm Levy, your PR reviewer

Good job so far :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- No linter error :heavy_check_mark: 
- Good UI :heavy_check_mark: 

## Status: REQUIRED CHANGES ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @levy002 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by amejid — APPROVED** [review]
Hi @seyioyemade and @AbbasSarwar, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- Linter checks passing :heavy_check_mark: 
- Correct workflow :heavy_check_mark: 
- React router used for navigation :heavy_check_mark: 
- Data fetching issue fixed :heavy_check_mark: 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**